docs: add camera streamer component example (#366)
aica-technology/api: docs/docs/examples/guides/camera-streamer.md

---
sidebar_position: 8
title: CameraStreamer component
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

import cameraStreamerExample from './assets/camera-streamer-example.png'

# CameraStreamer camera component

> AICA's `core-vision` package includes, among others, the `CameraStreamer` component that is responsible for reading

Reviewer: ```suggestion
AICA's `core-vision` package includes, among others, the `CameraStreamer` component that can read
```

image streams from USB devices, network streams, or video files. Most of the standard consumer-grade webcams and video
formats are supported out-of-the-box.

This guide describes shows the basics on how to use the component with a camera and a video file, two very common
choices.

## AICA Launcher configuration

> Start the AICA Launcher and add the `core-vision` package to your configuration at version v1.1.0. Mind that some

Author: @eeberhard as mentioned this morning. Do we go v4 or v5 here?

> Start the AICA Launcher and add the `core-vision` package to your configuration at version v1.1.0. Mind that some

Author: going v4

or by manually going under the `Core Vision Components` menu. Click to add it to the graph.
3. Next, connect the component to the start block.
4. Enable **auto-configure** and **auto-activate**
5. With your new component selected, make sure you are in the `Scene` menu, where you may find all the available
component parameters.

By this point, you should have something like the following:

<div class="text--center">
  <img src={cameraStreamerExample} alt="Default CameraStreamer configuration" />
</div>

Before pressing play, let us go through the parameters first. You should see:

- **Rate**: This is the component's rate, but it has no effect on the operation of `CameraStreamer`
- **Source**: Path to the source device or video file. If using a camera, this is typically of the form `/dev/videoX`,
whereas for video files you need to provide the absolute path to the video, e.g., `/path/to/video`
- **Camera frame**: The reference frame that will be used when publishing image messages, which should correspond the
camera's sensor position
- **Camera configuration**: A YAML-formatted camera configuration file containing the camera intrinsics (optional).

> <!-- TODO: If you don't have a calibration file for your camera, you can follow our [calibration guide](./camera-calibration.md) -->

Author: I'm actually tempted to say we should leave this link in (not in comment), target a small staging branch, and go to main as soon as the camera calibration PR is also merged.

> <!-- TODO: If you don't have a calibration file for your camera, you can follow our [calibration guide](./camera-calibration.md) -->

Reviewer: works for me
